File: src/types/point.ts
// 与 Rust models/point.rs 对齐
import type { ColorRGB, SampleConfig } from "./skill";

export interface Point {
  id: string;
  name: string;
  monitor: string;
  vx: number;
  vy: number;
  color: ColorRGB;
  tolerance: number;
  sample: SampleConfig;
  captured_at: string;
  note: string;
}

export interface PointsFile {
  schema_version: number;
  points: Point[];
}

File: src-tauri/src/models/point.rs
use serde::{Deserialize, Serialize};

use super::skill::{ColorRGB, SampleConfig};

/// 取色点位（对齐 python-legacy core/models/point.py）
#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct Point {
    pub id: String,
    pub name: String,
    pub monitor: String,
    pub vx: i32,
    pub vy: i32,
    pub color: ColorRGB,
    pub tolerance: u8,
    pub sample: SampleConfig,
    pub captured_at: String,
    pub note: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct PointsFile {
    pub schema_version: u32,
    pub points: Vec<Point>,
}
